Source/WebCore/platform/graphics/gstreamer/ImageDecoderGStreamer.cpp

```diff
@@ -94,6 +94,34 @@ ImageDecoderGStreamer::ImageDecoderGStreamer(FragmentedSharedBuffer& data, const
         GST_DEBUG_CATEGORY_INIT(webkit_image_decoder_debug, "webkitimagedecoder", 0, "WebKit image decoder");
     });
 
+    static Atomic<uint32_t> decoderId;
+    GRefPtr<GstElement> parsebin = gst_element_factory_make("parsebin", makeString("image-decoder-parser-", decoderId.exchangeAdd(1)).utf8().data());
+    m_parserHarness = GStreamerElementHarness::create(WTFMove(parsebin), [](auto&, const auto&) { }, [this](auto& pad) -> RefPtr<GStreamerElementHarness> {
+        auto caps = adoptGRef(gst_pad_query_caps(pad.get(), nullptr));
+        auto identityHarness = GStreamerElementHarness::create(adoptGRef(gst_element_factory_make("identity", nullptr)), [](auto&, const auto&) { });
+        GST_DEBUG_OBJECT(pad.get(), "Caps on parser source pad: %" GST_PTR_FORMAT, caps.get());
+        if (!caps || !doCapsHaveType(caps.get(), "video") || m_decoderHarness) {
+            GST_WARNING_OBJECT(m_decoderHarness->element(), "Parsed stream caps refused. Decoder already configured: %s", boolForPrinting(m_decoderHarness.get()));
```

Comment:
This message looks a bit confusing to understand.

Reply:
What's confusing?

Reply:
I'll split this in two warnings